test/integration/registration/spokecluster_autoapproval_test.go

```diff
@@ -88,4 +94,58 @@ var _ = ginkgo.Describe("Cluster Auto Approval", func() {
 			return nil
 		}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
 	})
+
+	ginkgo.It("Cluster can be denied by manually after automatically approved", func() {
+		clusterName := "autoapprovaltest-spoke-cluster1"
+		_, err := clusterClient.ClusterV1().ManagedClusters().Create(context.Background(), &clusterv1.ManagedCluster{
+			ObjectMeta: v1.ObjectMeta{
+				Name: clusterName,
+			},
+		}, v1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		gomega.Eventually(func() error {
+			cluster, err := util.GetManagedCluster(clusterClient, clusterName)
+			if err != nil {
+				return err
+			}
+
+			if _, ok := cluster.Annotations["open-cluster-management.io/initial-join-accepted"]; !ok {
```

Comment:
Based on this test, it seems that the purpose of the annotation is to approve the cluster automatically only once, so if the cluster is denied manually it will not approved automatically again. So the annotation can be renamed to "automatically-approved".

Reply:
may be used this annotation
`open-cluster-management.io/automaticlly-accpected-by-managedclusterctrl=<the accept timestamp>`